vllm/model_executor/layers/quantization/ipex_quant.py

```diff
@@ -70,32 +73,118 @@ def get_config_filenames() -> List[str]:
     @classmethod
     def from_config(cls, config: Dict[str, Any]) -> "IPEXConfig":
         method = cls.get_from_keys(config, ["quant_method"]).lower()
-        weight_bits = cls.get_from_keys(config, ["w_bit", "bits"])
-        group_size = cls.get_from_keys(config, ["q_group_size", "group_size"])
-        return cls(method, weight_bits, group_size)
+        if method == "awq":
+            weight_bits = cls.get_from_keys(config, ["w_bit", "bits"])
+            group_size = cls.get_from_keys(config,
+                                           ["q_group_size", "group_size"])
+            return cls(method, weight_bits, group_size, False, False)
+        if method == "gptq":
+            weight_bits = cls.get_from_keys(config, ["bits"])
+            group_size = cls.get_from_keys(config, ["group_size"])
+            lm_head_quantized = cls.get_from_keys_or(config, ["lm_head"],
+                                                     default=False)
+            try:
+                desc_act = cls.get_from_keys(config, ["desc_act"])
+            except Exception:
+                desc_act = False
+            return cls(method, weight_bits, group_size, desc_act,
+                       lm_head_quantized)
 
     @classmethod
     def override_quantization_method(cls, hf_quant_cfg,
                                      user_quant) -> Optional[str]:
-        if not current_platform.is_cpu():
+        if not current_platform.is_cpu() and not current_platform.is_xpu():
             return None
 
         quant_method = hf_quant_cfg.get("quant_method", "").lower()
 
-        if quant_method in ["awq"]:
+        if quant_method in ["awq", "gptq"]:
             return cls.get_name()
 
         return None
 
     def get_quant_method(self, layer: torch.nn.Module,
                          prefix: str) -> Optional["LinearMethodBase"]:
         if isinstance(layer, LinearBase):
-            return self.quant_method(self)
+            if self.method == "awq":
+                return IPEXAWQLinearMethod(self)
```

Comment:
Can you respect awq's modules_to_not_convert?
https://github.com/vllm-project/vllm/blob/de0e61a3239abff67c789138187a98465b806f76/vllm/model_executor/layers/quantization/awq_marlin.py#L123-L124